components/Wallet/WalletCard.tsx

```diff
@@ -5,50 +5,54 @@ import Image from 'next/image'
 import XECIcon from 'assets/xec-logo.png'
 import BCHIcon from 'assets/bch-logo.png'
 import EditWalletForm from './EditWalletForm'
+import { WalletWithAddressesAndPaybuttons, WalletPaymentInfo } from 'services/walletService'
 
 interface IProps {
-  walletInfo: {}
+  wallet: WalletWithAddressesAndPaybuttons
+  paymentInfo: WalletPaymentInfo
 }
-export default ({ walletInfo }: IProps): FunctionComponent => {
+
+export default ({ wallet, paymentInfo }: IProps): FunctionComponent => {
+  const networks = wallet.addresses.map((addr) => addr.networkId)
   return (
     <div className={style.wallet_card}>
       <div className={style.wallet_card_header_ctn}>
         <div className={style.wallet_card_header}>
-          <h4>{walletInfo.name}</h4>
+          <h4>{wallet.name}</h4>
           <div className={style.walletcard_icons}>
-            {walletInfo.xec_balance > 0 && <div><Image src={XECIcon} alt='XEC' /></div>}
-            {walletInfo.bch_balance > 0 && <div><Image src={BCHIcon} alt='BCH' /></div>}
+            {networks.includes(1) && <div><Image src={XECIcon} alt='XEC' /></div>}
```

Comment:
I'd feel better if we had a more reliable check than `networkId === 1` to indicate the currency of an address.
The client shouldn't have to know network ID's

Reply:
This isn't resolved yet. Just wanted to re-open this in case you missed it.

Reply:
Indeed I had missed it, thanks. Solved.